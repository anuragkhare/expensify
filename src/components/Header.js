import React from "react";
import { Link } from "react-router-dom";

const Header = () => {
  const subpages = [
    {
      to: "/",
      name: "Home"
    },
    {
      to: "/create",
      name: "Create"
    },
    // {
    //   to: "/edit",
    //   name: "Edit"
    // },
    {
      to: "/help",
      name: "Help"
    }
  ];

  return (
    <div>
      <h1>Expensify</h1>
      {subpages.map(({ to, name }, i) => (
        <span key={i}>
          <Link to={to}>{name}</Link>{" "}
        </span>
      ))}
    </div>
  );
};

export default Header;
