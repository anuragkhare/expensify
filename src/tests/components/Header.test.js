import React from "react";
// import ReactShallowRenderer from "react-test-renderer/shallow";
import { shallow } from "enzyme";
import Header from "../../components/Header";

test("Should render Header correctly", () => {
  // const renderer = new ReactShallowRenderer();
  // renderer.render(<Header />);
  // expect(renderer.getRenderOutput()).toMatchSnapshot();
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});
