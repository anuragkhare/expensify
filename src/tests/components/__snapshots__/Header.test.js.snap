// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<div>
  <h1>
    Expensify
  </h1>
  <span
    key="0"
  >
    <Link
      replace={false}
      to="/"
    >
      Home
    </Link>
     
  </span>
  <span
    key="1"
  >
    <Link
      replace={false}
      to="/create"
    >
      Create
    </Link>
     
  </span>
  <span
    key="2"
  >
    <Link
      replace={false}
      to="/help"
    >
      Help
    </Link>
     
  </span>
</div>
`;
